File: 0268-missing-number/solution.java
class Solution {
    public int missingNumber(int[] nums) {
         int s=0;int end=nums.length-1;
        int ans=nums.length;
        while(s<=end) {
           int correct=nums[s];
           
            if(nums[s]<=end && nums[s]!=nums[correct]){
                int temp=nums[s];
                nums[s]=nums[correct] ;
                nums[correct] =temp;
            } 
            else 
                s++;
        } 
        for(int i=0;i<nums.length;i++)
            if(nums[i]!=i)
                return i;
         
        return ans;
    }
}

File: 0268-missing-number/solution.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int i=0, n = nums.size();

        while(i < n){

            if(nums[i] == n || nums[i] == i) i++;
            else{
                int correctInd = nums[i];
                swap(nums[i], nums[correctInd]);
            }
        }

        for(int i=0; i<n; i++){
            if(nums[i] != i) return i;
        }
        return n;
    }
};
